src/pasta/curves.rs

```diff
@@ -796,14 +827,61 @@ macro_rules! impl_projective_curve_specific {
                 z: z3,
             };
 
-            $name::conditional_select(&tmp, &$name::zero(), self.is_zero())
+            $name::conditional_select(&tmp, &$name::identity(), self.is_identity())
+        }
+    };
+}
+
+macro_rules! impl_projective_curve_ext {
+    ($name:ident, $name_affine:ident, $iso:ident, $base:ident, special_a0_b5) => {
+        fn hash_to_curve<'a>(domain_prefix: &'a str) -> Box<dyn Fn(&[u8]) -> Self + 'a> {
+            use super::hashtocurve;
+
+            Box::new(move |message| {
+                let mut us = [Field::zero(); 2];
+                hashtocurve::hash_to_field($name_affine::CURVE_ID, domain_prefix, message, &mut us);
```

Comment:
Does it make more sense for `CURVE_ID` to be defined on `CurveExt`? Then this macro would not require `$name_affine`.

(I think we'll also remove `BLAKE2B_PERSONALIZATION` when we switch to using SSWU for the Halo bases; I just wanted to do that separately.)

Reply:
Oh, I guess this has already merged. We can make that simplification when we switch the Halo bases.